ADR/adr-0005-image-builder-sa.md

```diff
@@ -0,0 +1,52 @@
+# 5. Service Account configuration for image builds in the Pipeline Service 
+
+Date: Oct 5, 2022
+
+## Status
+
+Accepted
+
+## Context
+
+The `serviceAccount` used for running a `PipelineRun` workload in a workload cluster needs to have the right SCCs and linked credentials to be able to successfully execute a container image build. The linked credentials will mostly originate from the KCP workspace where the `PipelineRun` was synced from.
+Additionally, the SCC association needed for elevating capabilities required for image builds isn't something we'd prefer to expose the user to.
+
+
+
+## Decision
+
+* As a Pipeline Service platform feature, any service/user intending to build a container image would need to create and use the "pipeline-service-image-builder" service account in the KCP workspace. 
+Use of any other service accounts will not lead to a successful assembling of the image.
+
+* The AppStudio Build controller would tap into this platform contract to create relevant PipelineRuns for building container images.
+
+* The Pipeline Service Platform would configure the "pipeline-service-image-builder" service account in the execution environment ( ie, the workload cluster ) out-of-the-box.
+
+## Consequences
+
+
+
+### Implementation - Pipeline Service 
+
+* The contract/API between users and the platform is the "pipeline-service-image-builder" `serviceAccount` which consumers would be expected to use in the KCP workspace 
+for `PipelineRuns` if they wish to build container images. 
+* The Pipeline Service platform would configure workload clusters with the `[cluster]rolebindings` necessary to have the "pipeline-service-image-builder" `serviceAccount`. The infrastructure machinery driving the deployment of the `clusterrole`/`clusterrolebinding` should be capable of upgrading the privileges being assigned.
+configured with a tailor-made SCC such that they are sufficiently capable of building container images.
+* The service account does not necesarily need to be created in advance in the workload cluster by the Pipeline Service platform. Consumers of the Pipeline Service ( as specified in the next section ) would create this service account in the KCP workspace and have the same synced into the workload cluster's namespace. When the service account does get synced, the `[cluster]rolebindings` configured should kick-in to privide the appropriate elevation of privilege.
+
+
+### Implementation - AppStudio Build Service Controller
+
+* The AppStudio Build controller creates an "pipeline-service-image-builder" service account in the KCP workspace/namespace. This service account needs to be created so that it's made available in the workspace for linking the secrets in the subsequent step. 
```

Comment:
non-AppStudio/HACBS users will have to create this account manually. How about Pipeline Service creating the account and an empty secret, so that the user/controller only has to update the secret to have a working environment?

Reply:
Makes sense - would this be similar to your recommendation on setting up the service account on the kcp workspace for jvm-build-service users ?

Reply:
I think it is. We would provide 3 default users:
- `pipeline`: default account created by the tekton pipelines. Can be used for the most simple workloads.
- `pipeline-service-runner`: linked to KCP workspace. To be used when access to workspace CRs is required. Same permission levels as `pipeline`
- `pipeline-service-image-builder`: linked to KCP workspace, extra privileges on the workload cluster.